### PR TITLE
Replace handbuilt html with jelly for flow durability

### PR DESCRIPTION
**Pull request by timja**
The downside I can see with this is that tooltips are no longer shown.

But tooltips on a drop down are really weird. So not sure if it's the worst idea to get rid of this.

alternative to https://github.com/jenkinsci/workflow-api-plugin/pull/202

### CODE REVIEW

**Review by janfaracik — APPROVED** [review]

**Comment by timja** [comment]
> > But tooltips on a drop down are really weird. So not sure if it's the worst idea to get rid of this.
> 
> Maybe it's place a help button below the dropdown you can hover over to get some more intel about the mode selected? Could be beneficial for less versed pipeline users.

Can’t see how that would look good, there’s already some help for the input itself which should answer most questions

**Comment by jglick** [comment]
You can use `FormValidation.ok` (or `.okWithHtml`) to offer some descriptions of the current selection.

**Comment by jglick** [comment]
Merge conflicts @timja.

Do you have time to try https://github.com/jenkinsci/workflow-api-plugin/pull/205#issuecomment-1113625208? If not, I guess this is OK as is.

File: src/main/resources/org/jenkinsci/plugins/workflow/flow/GlobalDefaultFlowDurabilityLevel/global.jelly [hunk 1]
```diff
@@ -2,15 +2,7 @@
 <j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
     <f:section title="Pipeline Speed/Durability Settings">
         <f:entry title="${%Pipeline Default Speed/Durability Level}" field="durabilityHint">
-            <select class="setting-input" name="durabilityHint">
-                <f:option value="null" selected="${instance[field] == null}">None: use pipeline default (<j:out value="${descriptor.suggestedDurabilityHint.name()}" />)</f:option>
-                <j:forEach var="it" items="${descriptor.durabilityHintValues}">
```
Comment (jglick) [inline src/main/resources/org/jenkinsci/plugins/workflow/flow/GlobalDefaultFlowDurabilityLevel/global.jelly:7]: These methods could now be made private, I think, or simply inlined: https://github.com/jenkinsci/workflow-api-plugin/blob/786bbb37b24c7b0393f72a602547a8a85640d3e8/src/main/java/org/jenkinsci/plugins/workflow/flow/GlobalDefaultFlowDurabilityLevel.java#L82-L88

**Review by jglick — COMMENTED** [review]
I also just noticed that this whole thing should have been extending `GlobalConfiguration`. Probably too late to change that compatibly, though.

**Review by jglick — APPROVED** [review]

**Pull request merged**